mqtt/mqtt-broker/tests/broker_model.rs

```diff
@@ -0,0 +1,240 @@
+use std::collections::{HashMap, HashSet};
+
+use proptest::{prop_oneof, proptest, strategy::Strategy};
+use tokio::sync::mpsc::{self, UnboundedReceiver};
+use uuid::Uuid;
+
+use mqtt3::proto;
+use mqtt_broker::{
+    proptest::{arb_client_id_weighted, arb_connect, arb_subscribe, arb_unsubscribe},
+    AuthId, BrokerBuilder, ClientEvent, ClientId, ConnReq, ConnectionHandle, Message,
+};
+
+proptest! {
+    #[test]
+    fn broker_manages_sessions(
+        events in proptest::collection::vec(arb_broker_event(), 1..50)
+    ) {
+    tokio::runtime::Builder::new()
+        .basic_scheduler()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(test_broker_manages_sessions(events));
+    }
+}
+
+async fn test_broker_manages_sessions(events: impl IntoIterator<Item = BrokerEvent>) {
+    let mut broker = BrokerBuilder::default()
+        .authenticator(|_| Ok(Some(AuthId::Anonymous)))
+        .authorizer(|_| Ok(true))
+        .build();
+
+    let mut model = BrokerModel::default();
+
+    let mut clients = Vec::new();
```

Comment:
I'm not sure I understand `clients` vector and what it is used for? Is it just to keep receivers from dropping?

Reply:
yep, you're right. when we include message dispatching it will collect messages for each client.